utils/general.py

```diff
@@ -36,6 +38,73 @@
 # Prevent OpenCV from multithreading (to use PyTorch DataLoader)
 cv2.setNumThreads(0)
 
+# Download objects
+def download_file(dir, blob):
+    file_name = blob.name.split("/")[2]
+    destination = Path(dir) / f"{file_name}"
+    destination.touch()
+    blob.download_to_filename(destination)
+
+
+# Place GCS objects in their respective local directories
+def place_object(local_dir, blob):
+    image_train = "images/t"
+    image_val = "images/v"
+    label_train = "labels/t"
+    label_val = "labels/v"
+    jpg = ".jpg"
+    txt = ".txt"
+
+    if blob.name.startswith(image_train) and blob.name.endswith(jpg):
+        download_file(local_dir[0], blob)
+
+    elif blob.name.startswith(label_train) and blob.name.endswith(txt):
+        download_file(local_dir[1], blob)
+
+    elif blob.name.startswith(image_val) and blob.name.endswith(jpg):
+        download_file(local_dir[2], blob)
+
+    elif blob.name.startswith(label_val) and blob.name.endswith(txt):
+        download_file(local_dir[3], blob)
+
+    else:
+        pass
+
+ 
+# Decorator for "check_dataset"
+def check_decorator(func):
```

Comment:
I would rename this decorator something that tells us what it does instead. Something alluding to the fact that we are training a YOLO network, since that assumes a certain folder structure.

Generally you should try not to name things after their implementation or technical role in the code, because those might change.

Reply:
Fixed a33490a6a6b0d8860a9e67280bf0972a23f022c3